org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/ex/ExternalAntlrLexerFragment.java

```diff
@@ -0,0 +1,361 @@
+		// i am not sure when to generate what
+		if (runtime) {
+			StringConcatenationClient binding = new StringConcatenationClient() {
+				@Override
+				protected void appendTo(TargetStringConcatenation target) {
+					target.append("binder.bind(" + Lexer.class.getName() + ".class)"+
+							".annotatedWith(com.google.inject.name.Names.named(" +
+							"org.eclipse.xtext.parser.antlr.LexerBindings.RUNTIME" +
+							")).to(" + lexerGrammar +".class);");
+				}
+			};
+			new GuiceModuleAccess.BindingFactory().addConfiguredBinding("RuntimeLexer", binding).contributeTo(getLanguage().getRuntimeGenModule());
+		}
+		if (highlighting) {
+				if (this.getProjectConfig().getEclipsePlugin().getRoot() != null) {
+					StringConcatenationClient binding = new StringConcatenationClient() {
+						@Override
+						protected void appendTo(TargetStringConcatenation target) {
+							target.append("binder.bind(" + Lexer.class.getName() + ".class)"+
+									".annotatedWith(com.google.inject.name.Names.named(" +
+									"org.eclipse.xtext.ide.LexerIdeBindings.HIGHLIGHTING" +
+									")).to(" + lexerGrammar +".class);");
+						}
+					};
+					new GuiceModuleAccess.BindingFactory().addConfiguredBinding("HighlightingLexer", binding).contributeTo(getLanguage().getEclipsePluginGenModule());
+				}
+			}
+			if (contentAssist) {
+				if (this.getProjectConfig().getGenericIde().getRoot() != null) {
+					StringConcatenationClient binding = new StringConcatenationClient() {
+						@Override
+						protected void appendTo(TargetStringConcatenation target) {
+							target.append("binder.bind(org.eclipse.xtext.ide.editor.contentassist.antlr.internal.Lexer.class)"+
+									".annotatedWith(com.google.inject.name.Names.named(" +
+									"org.eclipse.xtext.ide.LexerIdeBindings.CONTENT_ASSIST" +
+									")).to(" + lexerGrammar +".class);");
+						}
+					};
+					new GuiceModuleAccess.BindingFactory().addConfiguredBinding("ContentAssistLexer", binding).contributeTo(getLanguage().getIdeGenModule());
+			
+				} 
+				if (this.getProjectConfig().getEclipsePlugin().getRoot() != null) {
+					StringConcatenationClient binding = new StringConcatenationClient() {
+						@Override
+						protected void appendTo(TargetStringConcatenation target) {
+							target.append("binder.bind(org.eclipse.xtext.ide.editor.contentassist.antlr.internal.Lexer.class)"+
+									".annotatedWith(com.google.inject.name.Names.named(" +
+									"org.eclipse.xtext.ide.LexerIdeBindings.CONTENT_ASSIST" +
+									")).to(" + lexerGrammar +".class);");
+						}
+					};
+					new GuiceModuleAccess.BindingFactory().addConfiguredBinding("ContentAssistLexer", binding).contributeTo(getLanguage().getEclipsePluginGenModule());
+				}
+			}
+			
+			
+		
+		
+		IXtextGeneratorFileSystemAccess srcGenFsa = this.getProjectConfig().getRuntime().getSrcGen();
+		IXtextGeneratorFileSystemAccess srcFsa = this.getProjectConfig().getRuntime().getSrc();
+		if (contentAssist || highlighting) {
```

Comment:
am not sure about this. in ide we have highlighting only correct?

Reply:
We do generate to `org.xtext.example.mydsl.ide.contentassist.antlr`, so we do have CA in IDE. Highlighting should only be in UI, since it is platform specific.